### DISCUSSION TITLE
How to get older types for Pokemon?

### DISCUSSION THREAD

**Discussion by antalest** [PokeAPI/pokeapi#803] [Q&A]
For example Clefairy used to be Normal type. Nowadays it's Fairy type and that's what I get from PokeAPI. Is there a way to get types (and other data, perhaps stats have changed too) for each pokemon as they used be in specific game versions?

EDIT:
Oh... I just found about the past_types :D The readme in this repo specifically has Clefairy as an example. So if that property exists in response and my generation is older or the same, I can use that value instead. A bit complicated but doable.

Then I wonder if there is any endpoint to query game version / generation specific pokemon data directly?

**Comment by Naramsim** [comment]
No sorry, not really the API wasn't thought to support older values. The `past_types` was a sort of hack added by a contributor.

**Reply by antalest** [reply]
Okay, thanks for the reply though :)

Would you happen to know any resource with this kind of data available?

**Reply by yazae** [reply]
Well, serebii for exemple https://www.serebii.net/pokedex-swsh/clefairy/ has a pokedex by generation where you can find those data, but idk about a specific endpoint for pokemon changes by version in english.
